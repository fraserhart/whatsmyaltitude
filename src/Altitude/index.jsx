import React from 'react';
import PropTypes from 'prop-types';
import { ERROR_GETTING_LOCATION, WAITING_FOR_LOCATION } from '../Constants';

const Altitude = ({ altitude }) => {
  const formattedAltitude = () => {
    if (altitude !== WAITING_FOR_LOCATION && altitude !== ERROR_GETTING_LOCATION){
      const meterNumber = Number(altitude);
      const feetNumber = meterNumber * 3.28;
      return `${meterNumber.toFixed(2)} meters / ${feetNumber.toFixed(2)} feet`;
    }
    return altitude;
  };
  return (
    <h2>
      { formattedAltitude() }
    </h2>
  );
};

Altitude.defaultProps = {
  altitude: WAITING_FOR_LOCATION,
};

Altitude.propTypes = {
  altitude: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
};

export default Altitude;