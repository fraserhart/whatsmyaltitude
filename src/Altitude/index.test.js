import React from 'react';
import renderer from 'react-test-renderer';
import Altitude from '.';

it('displays the formatted altitude', () => {
  const wrapper = renderer.create(<Altitude altitude={5.6} />);
  expect(wrapper).toMatchSnapshot();
})
