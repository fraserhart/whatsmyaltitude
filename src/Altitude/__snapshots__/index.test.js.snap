// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays the formatted altitude 1`] = `
<h2>
  5.60 meters / 18.37 feet
</h2>
`;
